### ISSUE TITLE
Network graph: bind nodes

### ISSUE THREAD

**Issue by mepaling** [almende/vis#2002]
Hello, I'm new to vis.js.
Hope I won't say anything to offend you due to my poor English. 

As the title, I'm trying to bind nodes.
The graph I made is a network topology. It contains IP node and Port node.
If a port belongs to an IP, I use an edge to connect both nodes.
The data flow is an edge between two ports.
[An Example here](https://jsbin.com/natavacona/1/edit?html,output)
What I really want(i.e. binding nodes) is like the graph following:
![123](https://cloud.githubusercontent.com/assets/10512271/17356807/0af1c5b6-598c-11e6-8b71-362b0e1319b4.png)

PS. I use hierarchical layout because it meets my need.
Thank you!

**Comment by controversial** [comment]
This isn't currently possible easily. You could look into disconnecting a node from the graph by eliminating edges to it, then positioning it with `fixed` and directly below the other node. You'd then have to listen for changes to the node's position with the [`dragging` event](http://visjs.org/docs/network/#Events) and update the position of the fixed node.

But there isn't a convenient built-in way to do this right now.

**Comment by mepaling** [comment]
@controversial Sorry for the late reply and thank you for your response, but I finally switched to d3.js. Now I think I should close the issue.

**Comment by controversial** [comment]
Great, glad you resolved it.

**Comment by no1youknowz** [comment]
@mepaling I'm currently looking for the same effect, how did you achieve it?  Did you see an existing example over at d3?  Thanks

**Comment by mojoaxel** [comment]
> But there isn't a convenient built-in way to do this right now.

@mepaling @no1youknowz Feel free to reopen this. We could change it to a "feature-request" if this is still needed.

**Comment by mepaling** [comment]
@no1youknowz 
Sorry for the late reply. I'm busy on another project.
And please don't mind with my poor English.
I have been away of these things for several months. 
Now I'm try to read my old codes about these things.

I'm also new to D3.js, so there may have a better way to achieve the goal.
Sorry for not providing codes because those things are parts of an enterprise's project.

The following operations is that how I achieve this:
First, I name these component as IPnode (ex. 192.168.1.1) and PortNode(ex. 12345) and LINK (the line between two nodes).
1. I use default d3.forceSimulation() to get the data ready. (for LINES d.x, d.y)
2. Draw the LINK and IPnode by the data( d.x, d.y )
3. Override the end function (i.e. d3.end("tick", tickisdone)) to add and draw PortNodes
In conclusion:
The trick is knowing the LINK's start position and end position. Then just adjust the nodes' x and y base on these positions.
Finally, remember the component which draws next will overlap the last component of the same positions.

@mojoaxel 
I've left the project which is related to this topic, so I think I won't open the issue. If someone think the feature is still needed, then just open another feature-request issue.

Thank you!

**Comment by no1youknowz** [comment]
@mepaling thanks for your answer and your english was just fine! 👍 

@mojoaxel Whilst I have an existing implementation with this library, I have decided to go with jsplumb tooklit.
